src/com/facebook/buck/zip/JarBuilder.java

```diff
@@ -235,6 +254,20 @@ private void addEntryToJar(JarEntrySupplier entrySupplier, CustomJarOutputStream
       return;
     }
 
+    // Collect all services together for later merging and addition to the output jar
+    if (isService(entryName)) {
+      try (InputStream entryInputStream = entrySupplier.getInputStreamSupplier().get()) {
+        if (entryInputStream != null) {
+          // Null stream means a directory
+          Set<String> existingServices = services.getOrDefault(entryName, new LinkedHashSet<>());
```

Comment:
Use of `getOrDefault` here is going to create a garbage `LinkedHashSet` every time. `computeIfAbsent` is better. You also won't need the `putIfAbsent` afterwards.